packages/venia-ui/lib/util/makeUrl.js

```diff
@@ -1,8 +1,14 @@
+import { BrowserPersistence } from '@magento/peregrine/lib/util';
+
+const storage = new BrowserPersistence();
+
 // If the root template supplies the backend URL at runtime, use it directly
 const htmlDataset = document.querySelector('html').dataset;
 const { imageOptimizingOrigin } = htmlDataset;
 // Protect against potential falsy values for `mediaBackend`.
-let mediaBackend = htmlDataset.mediaBackend;
+let mediaBackend =
+    storage.getItem('store_view_secure_base_media_url') ||
```

Comment:
This will only fetch the stored value on initial eval. You'd need to call `getItem` in the functions that reference `mediaBackend` so that the value is always fresh.

That said, I'm not sure what the affect will be on performance if we start doing storage reads for EVERY image/url.

Reply:
Decided together that we would do a storage read, and fallback to global data from `AVAILABLE_STORE_VIEWS`, to build the `mediaBackend` url.